reactive-banana/src/Reactive/Banana/Prim/Graph.hs

```diff
@@ -47,9 +65,9 @@ getParents gr x = maybe [] id . Map.lookup x . parents $ gr
 listParents :: forall a. (Eq a, Hashable a) => Graph a -> [a]
 listParents gr = list
     where
-    -- all nodes without children
+    -- all nodes without parents
```

Comment:
I'm pretty sure this was just a confusing typo, and not a bug in the implementation. We _keep_ nodes that have no parents with this computation:

```haskell
[ x | x <- Set.toList $ nodes gr, null (getParents gr x)]
```